oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/config/annotation/web/configurers/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -210,6 +210,18 @@ public OAuth2AuthorizationServerConfigurer tokenRevocationEndpoint(Customizer<OA
 		return this;
 	}
 
+	/**
+	 * Configures the OAuth 2.0 Device Authorization Endpoint.
+	 *
+	 * @param deviceAuthorizationEndpointCustomizer the {@link Customizer} providing access to the {@link OAuth2DeviceAuthorizationEndpointConfigurer}
+	 * @return the {@link OAuth2AuthorizationServerConfigurer} for further configuration
+	 * @since 1.1
+	 */
+	public OAuth2AuthorizationServerConfigurer deviceAuthorizationEndpoint(Customizer<OAuth2DeviceAuthorizationEndpointConfigurer> deviceAuthorizationEndpointCustomizer) {
+		deviceAuthorizationEndpointCustomizer.customize(getConfigurer(OAuth2DeviceAuthorizationEndpointConfigurer.class));
```

Comment:
Perhaps, if it's similar to lowering default surface area. I also wonder if the standard `invalid_client` error (or `invalid_grant` error, can't remember which...) because the grant type is not allowed for a particular `RegisteredClient` cover this case?